File: bacha.2022.03.30/6.cpp
#define rep(i, n) for (int i = 0; i < (int)(n); ++i)
#define reps(i, s, n) for (int i = (s); i < (int)(n); ++i)
#define rrep(i, n) for (int i = (int)(n-1); i >= 0; --i)
#define MOD 1000000007
#define INF 2000000000
#define INFL 1000000000000000000
template <typename T> bool chmin(T &a, const T& b) { if (a > b) { a = b; return true; } return false; }
template <typename T> bool chmax(T &a, const T& b) { if (a < b) { a = b; return true; } return false; }
template <typename T> int count_digit(T n) { int res = 0; while (n > 0) { ++res; n /= 10; } return res; }
typedef long long ll;
#include <bits/stdc++.h>
using namespace std;

int main() {
    string s, t;
    cin >> s >> t;
    int sn = s.length();
    int tn = t.length();
    int ansn = 0;
    string ans = "";

    //vector<vector<int>> dp(sn+1, vector<int>(tn+1, 0));
    rep (i, sn) {
        vector<vector<int>> dp(sn+1, vector<int>(tn+1, 0));
    }

    cout << ans << endl;
    return 0;
}

File: bacha/bacha.2022.06.14/6.go
package main

import (
    "fmt"
)

const MOD = 998244353

func main() {
    var n int
    fmt.Scanf("%d", &n)
    dp := make([][]int, n+1)
    for i := int(0); i < n+1; i++ {
        dp[i] = make([]int, 9)
    }
    for i := 0; i < 9; i++ {
        dp[0][i] = 1
    }

    for i := 1; i < n; i++ {
        for j := 0; j < 9; j++ {
            if 0 <= j-1 {
                dp[i][j] = (dp[i][j] + dp[i-1][j-1]) % MOD
            }
            if j+1 < 9 {
                dp[i][j] = (dp[i][j] + dp[i-1][j+1]) % MOD
            }
            dp[i][j] = (dp[i][j] + dp[i-1][j]) % MOD
        }
    }

    ans := 0
    for i := 0; i < 9; i++ {
        ans = (ans + dp[n-1][i]) % MOD
    }
    fmt.Println(ans)
}
